dashboard/app/controllers/openai_chat_controller.rb

```diff
@@ -10,9 +10,11 @@ def chat_completion
 
     # Check for PII / Profanity
     locale = params[:locale] || "en"
-    messages = params[:messages].collect {|msg| msg[:content]}.join(' ')
-    filter_result = ShareFiltering.find_failure(messages, locale) if messages
-    return render(status: :bad_request, json: {type: filter_result.type, content: filter_result.content}) if filter_result
+    # Just look at the most recent message from the student.
+    message = params[:messages].last[:content]
```

Comment:
Why do we only need to look at the most recent message?

Reply:
Any previous messages will have already been flagged and displayed as such in the UI. If we look for filter failures in the whole set of messages every time, then appropriate messages that come after inappropriate ones will be incorrectly flagged as inappropriate.